tests/components/august/mocks.py

```diff
@@ -39,44 +39,123 @@ def _mock_get_config():
 
 @mock.patch("homeassistant.components.august.Api")
 @mock.patch("homeassistant.components.august.Authenticator.authenticate")
-async def _mock_setup_august(hass, api_mocks_callback, authenticate_mock, api_mock):
+async def _mock_setup_august(hass, api_instance, authenticate_mock, api_mock):
     """Set up august integration."""
     authenticate_mock.side_effect = MagicMock(
         return_value=_mock_august_authentication("original_token", 1234)
     )
-    api_mocks_callback(api_mock)
+    api_mock.return_value = api_instance
     assert await async_setup_component(hass, DOMAIN, _mock_get_config())
     await hass.async_block_till_done()
     return True
 
 
-async def _create_august_with_devices(hass, lock_details=[], doorbell_details=[]):
-    locks = []
-    doorbells = []
-    for lock in lock_details:
-        if isinstance(lock, LockDetail):
-            locks.append(_mock_august_lock(lock.device_id))
-    for doorbell in doorbell_details:
-        if isinstance(lock, DoorbellDetail):
-            doorbells.append(_mock_august_doorbell(doorbell.device_id))
-
-    def api_mocks_callback(api):
-        def get_lock_detail_side_effect(access_token, device_id):
-            for lock in lock_details:
-                if isinstance(lock, LockDetail) and lock.device_id == device_id:
-                    return lock
-
-        api_instance = MagicMock()
-        api_instance.get_lock_detail.side_effect = get_lock_detail_side_effect
-        api_instance.get_operable_locks.return_value = locks
-        api_instance.get_doorbells.return_value = doorbells
-        api_instance.lock.return_value = LockStatus.LOCKED
-        api_instance.unlock.return_value = LockStatus.UNLOCKED
-        api.return_value = api_instance
-
-    await _mock_setup_august(hass, api_mocks_callback)
-
-    return True
+async def _create_august_with_devices(hass, devices, api_call_side_effects={}):
```

Comment:
We can default `api_call_side_effects` to `None` and set it to an empty dict if `None` in the body of the function instead.

Reply:
done